server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -348,12 +352,28 @@ public Page<FlowInstanceEntity> listAll(@NotNull Pageable pageable, @NotNull Que
         Specification<FlowInstanceViewEntity> specification = Specification
                 .where(FlowInstanceViewSpecs.creatorIdIn(creatorIds))
                 .and(FlowInstanceViewSpecs.organizationIdEquals(authenticationFacade.currentOrganizationId()))
-                .and(FlowInstanceViewSpecs.taskTypeEquals(params.getType()))
                 .and(FlowInstanceViewSpecs.projectIdEquals(params.getProjectId()))
                 .and(FlowInstanceViewSpecs.statusIn(params.getStatuses()))
                 .and(FlowInstanceViewSpecs.createTimeLate(params.getStartTime()))
                 .and(FlowInstanceViewSpecs.createTimeBefore(params.getEndTime()))
                 .and(FlowInstanceViewSpecs.idEquals(targetId));
+        if (params.getType() != null) {
```

Comment:
In 4.2.0, @PeachThinking create a VIEW  named `list_flow_instance_view` in metaDB to solve the issues about list flow instance. In this view, the task type is hardcode, so that we cannot list a new type of task. So in this PR, I make it codeable by deleting hardcode in view's DDL  and adding JPA specification condition.